Frameworks/UIKit/NSString+UIKitAdditions.mm

```diff
@@ -161,48 +88,78 @@ - (void)drawInRect:(CGRect)rect withAttributes:(NSDictionary*)attrs {
 /**
  @Status Interoperable
 */
-- (CGSize)drawInRect:(CGRect)rct withFont:(UIFont*)font lineBreakMode:(UILineBreakMode)lineBreakMode alignment:(UITextAlignment)alignment {
-    CGSize fontExtent;
-
-    drawString(font, UIGraphicsGetCurrentContext(), self, rct, lineBreakMode, alignment, &fontExtent);
-
-    return fontExtent;
+- (CGSize)drawInRect:(CGRect)rect withFont:(UIFont*)font {
+    return [self drawInRect:rect withFont:font lineBreakMode:UILineBreakModeWordWrap];
 }
 
 /**
  @Status Interoperable
 */
-- (CGSize)drawInRect:(CGRect)rct withFont:(UIFont*)font lineBreakMode:(UILineBreakMode)lineBreakMode {
-    CGSize fontExtent;
-
-    drawString(font, UIGraphicsGetCurrentContext(), self, rct, lineBreakMode, UITextAlignmentLeft, &fontExtent);
-
-    return fontExtent;
+- (CGSize)drawInRect:(CGRect)rect withFont:(UIFont*)font lineBreakMode:(UILineBreakMode)lineBreakMode {
+    return [self drawInRect:rect withFont:font lineBreakMode:lineBreakMode alignment:UITextAlignmentLeft];
 }
 
 /**
  @Status Interoperable
 */
-- (CGSize)drawAtPoint:(CGPoint)pt withFont:(UIFont*)font {
-    CGSize fontExtent;
+- (CGSize)drawInRect:(CGRect)rect withFont:(UIFont*)font lineBreakMode:(UILineBreakMode)lineBreakMode alignment:(UITextAlignment)alignment {
+    CGContextRef context = UIGraphicsGetCurrentContext();
+    if (rect.size.width == 0) {
+        rect.size.width = std::numeric_limits<CGFloat>::max();
+    }
+
+    if (rect.size.height == 0) {
+        rect.size.height = std::numeric_limits<CGFloat>::max();
+    }
 
-    CGRect rct;
+    CTParagraphStyleSetting styles[2];
+    CTTextAlignment align = _NSTextAlignmentToCTTextAlignment(alignment);
+    styles[0] = { kCTParagraphStyleSpecifierAlignment, sizeof(CTTextAlignment), &align };
 
-    rct.origin.x = pt.x;
-    rct.origin.y = pt.y;
-    rct.size.width = 0;
-    rct.size.height = 0;
+    CTLineBreakMode breakMode = static_cast<CTLineBreakMode>(lineBreakMode);
+    styles[1] = { kCTParagraphStyleSpecifierLineBreakMode, sizeof(CTLineBreakMode), &breakMode };
+
+    NSAttributedString* attr =
+        [[[NSAttributedString alloc] initWithString:self
+                                         attributes:@{
+                                             (NSString*)kCTForegroundColorFromContextAttributeName : (id)kCFBooleanTrue,
```

Comment:
>(NSString*) [](start = 45, length = 11)

switching to static cast is going to really obscure readability.
I guess we can leave it as be